src/LiveComponent/doc/index.rst

```diff
@@ -2661,6 +2678,125 @@ which will be used to identify each child component. You can
 also pass in a ``data-live-id`` attribute directly, but ``key`` is
 a bit more convenient.
 
+Tricks with a Loop + a "New" Item
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Let's get fancier. Suppose you're rendering all of the line items
+inside of an `InvoiceCreator` component. After the loop, you might
+decide to render one more component to create a *new* line item.
+In that case, you can pass in a ``key`` set to something like ``new_line_item``:
+
+.. code-block:: twig
+
+    {{ component('InvoiceLineItem', {
+        key: 'new_line_item',
+    }) }}
+
+Imagine you also have a ``LiveAction`` inside of ``InvoiceLineItem``
+that saves the new line item to the database. To be extra fancy,
+it emits a ``lineItem:created`` event to the parent::
+
+    // src/Twig/InvoiceLineItem.php
+    // ...
+
+    #[AsLiveComponent]
+    final class InvoiceLineItem
+    {
+        // ...
+
+        #[LiveProp]
+        #[Valid]
+        public InvoiceLineItem $lineItem;
+
+        #[PostMount]
+        public function postMount(): void
+        {
+            if(!$this->lineItem) {
+                $this->lineItem = new InvoiceLineItem();
```

Comment:
It's an inception? Component creates itself? :grin:

Reply:
Haha, whoops! I'll change the entity name - good catch